### PR TITLE
Add Swift Package Manager (SPM) example

### PR DESCRIPTION
**Pull request by 123FLO321**
Adds example configuration for [Swift Package Manager (SPM)](https://swift.org/package-manager/) based projects.

### CODE REVIEW

File: examples.md [hunk 1]
```diff
@@ -373,3 +374,14 @@ When dependencies are installed later in the workflow, we must specify the same
     restore-keys: |
       ${{ runner.os }}-pods-
 ```
+
+## Swift - Swift Package Manager
+
+```yaml
+- uses: actions/cache@v1
+  with:
+    path: .build
```
Comment (joshmgross) [inline examples.md]: Is saving the entire `.build` directory recommended?

Comment (123FLO321) [reply examples.md]: Swift places all build files directly in that folder.
That's also how IBMs swift-buildpack does caching (https://github.com/IBM-Swift/swift-buildpack#caching-of-the-build-directory)

**Comment by abinrm** [comment]
@123FLO321  Can you please help me with this condition to check if there is a cache is present or not. My `CI.yml` looks like below 

```
name: CI

on: [push]

jobs:
  build:

    runs-on: macos-latest

    steps:
    - name: Checkout
      uses: actions/checkout@v1
    
    - name: Cache
      uses: actions/cache@v1.1.2
      with:
        path: .build
        key: ${{ runner.os }}-spm-${{ hashFiles('**/Package.resolved') }}
        restore-keys: |
          ${{ runner.os }}-spm-
      
    - name: Build app
      if: steps.build.outputs.cache-hit != 'true'
      run: xcodebuild -project 'PROJECTNAME.xcodeproj' -scheme 'SCHEME' -destination 'generic/platform=tvOS' -configuration Release build CODE_SIGNING_ALLOWED=NO | xcpretty
```
Every time I get the following while running this GitHub action for cache
`Cache not found for input keys: macOS-spm-d5b57c01375b10bad0ebb66287afdf2cd3efa116fa0dacf95341e3e895592293, macOS-spm-.`

File: examples.md (hunk 1 above)
Comment (JosephDuffy) [reply examples.md:383]: @123FLO321 Have you found this caching works for built files or artefacts? https://bugs.swift.org/browse/SR-11760 is being used to track an issue related to how SwiftPM tracks files for the build cache. There was a recent PR for this that was not merged: https://github.com/apple/swift-package-manager/pull/2774.

In my testing (see https://github.com/JosephDuffy/actions-playground/runs/954106324?check_suite_focus=true vs https://github.com/JosephDuffy/actions-playground/runs/954129569?check_suite_focus=true) I see the checkouts cache being used, but the build cache is not being used at all. This is true on stable Xcode (11.5) as was as the 12 beta.

Comment (123FLO321) [reply examples.md:383]: @JosephDuffy the Problem there is that no Package.resolved file exists when trying to load the cache. Try adding a `swift package resolve` step before loading cache.

Comment (JosephDuffy) [reply examples.md:383]: @123FLO321 The checkout step has completed and there's a Package.resolved file in that branch (https://github.com/JosephDuffy/actions-playground/compare/swiftpm-cache#diff-d29466d627f33a3d61709bad13c5eadd). It's a cache hit (see the second link) but SwiftPM still performs a rebuild.

The links are a little long/unclear so I'll update the comment to be clearer.

**Pull request merged**